### PR TITLE
Fix centipede crash dir

### PR DESCRIPTION
**Pull request by Alan32Liu**
Copy crash reproducer to the `reproducers_dir` and fix unit tests accordingly.

### CODE REVIEW

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 1]
```diff
@@ -57,13 +58,15 @@ def _get_runner():
   return new_process.UnicodeProcessRunner(centipede_path)
 
 
-def _get_reproducer_path(log):
+def _get_reproducer_path(log, reproducers_dir):
   """Gets the reproducer path, if any."""
   crash_match = _CRASH_REGEX.search(log)
-  if not crash_match:
+  if not crash_match or not crash_match.group(1):
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:64]: Under what circumstance can the second part (`not crash_match.group(1)`) occur? Please add a comment.

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py [hunk 2]
```diff
@@ -57,13 +58,15 @@ def _get_runner():
...
     return None
-
-  return crash_match.group(1)
+  tmp_crash_path = Path(crash_match.group(1))
+  prm_crash_path = Path(reproducers_dir) / tmp_crash_path.name
```
Comment (oliverchang) [inline src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py:67]: just call this variable `crash_path`. Not clear what `prm` means here either.

**Review by oliverchang — APPROVED** [review]
thanks! some minor comments.

File: src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py (hunk 1 above)
Comment (Alan32Liu) [reply src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py]: Right, that part is a tautology given the regex is not optional.
Removed in the new commit.

**Pull request merged**